### ISSUE TITLE
Question: Dynamic String Configuration per Page and Component

### ISSUE THREAD

**Issue by nbouvrette** [vinissimus/next-translate#587]
I've been trying for a while to find a library where I could do the following:

- Each page would have its own JSON file (per language) and could only be used on that page.
- Each component would have its own JSON file (per language) and could only be used on that component.

Basically, strings would work similar to CSS and live close to the asset where they are used. 

I thought originally next-translate would be able to help me with this but from what I saw all the strings association happens at the page level and it would not be possible to achieve this result without major code change?

**Comment by aralroca** [comment]
@nbouvrette You can load at page level all translations that are visible when rendering, and load dynamic components dynamically with [DynamicNamespaces](https://github.com/vinissimus/next-translate/tree/1.0.6#dynamicnamespaces).

**Comment by nbouvrette** [comment]
@aralroca unless I misunderstood dynamic namespaces are not available for SSR? I should have mentioned in my original question that I am looking for a solution that will be part of the SSR HTML returned by Next.js.  This has been my big challenge with this idea.

I am starting to think I will need to explore some sort of custom Webpack loader option but I might be wrong.

**Comment by aralroca** [comment]
@nbouvrette yes, `DynamicNamespaces` [loads the translations in a `useEffect`](https://github.com/vinissimus/next-translate/blob/master/src/DynamicNamespaces.tsx#L30) so in SSR it will only load the spinner or skeleton. Next-translate is designed to control translations at the page level, not at the component level, so it is a limitation.

**Comment by nbouvrette** [comment]
@aralroca do you have any recommendations to achieve what I have in mind? I have been working on a router for Next.js that can translate URLs and this is the last bit of the solution I am trying to complete. I would be happy to show you more in detail if you are interested. My goal is to open source it once ready.

**Comment by aralroca** [comment]
@nbouvrette I wish I would help you, but I'm struggling to see how we could implement that each component would have its own JSON but at the same time if you use that component 100 times in a list, only load the JSON once at the page level. I guess it would be a loader similar to the CSS, but I'm struggling to see how to make it use the right language and it can change dynamically.

**Comment by nbouvrette** [comment]
@aralroca what I was picturing is exactly like [CSS modules](https://github.com/css-modules/css-modules) but being able to break each strings file into Webpack chunks, let Next.js load the right chunk during SSR and also after on the client-side. I have no idea if this is even possible but conceptually I think it would be a great solution to bind strings close to where they are used (just like CSS).

**Comment by aralroca** [comment]
I am with you, if something like this could be done, it would be really useful.

**Comment by nbouvrette** [comment]
@aralroca We've done it: https://github.com/Avansai/next-multilingual

Let me know what you think :)

**Comment by aralroca** [comment]
@nbouvrette It looks good, when I have more time I will try it and see how it is. And congratulations 😊
